File: challenge4/timeConversion.py
def timeConversion(s):
    # Write your code here
    isAM = s[-2:] == 'AM'
    if isAM:
        if s[:2] == '12':
            return '00' + s[2:-2]
        else:
            return s[:-2]
    strHour = s[:2]
    strRest = s[2:-2]
    hour = 12 if int(strHour) == 12 else int(strHour) + 12
    strHour = '00' if hour == 0 else str(hour)
    return strHour + strRest


if __name__ == '__main__':
    timeConversion('07:05:45AM') # 07:05:45
    timeConversion('12:00:00AM') # 00:00:00
    timeConversion('12:00:01AM') # 00:00:01
    timeConversion('12:00:00PM') # 12:00:00
    timeConversion('12:45:54PM') # 12:45:54
    timeConversion('07:05:45PM') # 19:05:45

File: challenge4/TimeConversion.java
package challenge4;

public class TimeConversion {
    public static String timeConversion(String s) {
        // Write your code here
        boolean isAM = s.endsWith("AM");
        if (isAM) {
            if (s.startsWith("12")) {
                return "00" + s.substring(2, 8);
            } else {
                return s.substring(0, 8);
            }
        }
        String strHour = s.substring(0, 2);
        String strRest = s.substring(2, 8);
        int hour = Integer.parseInt(strHour);
        hour = hour == 12 ? 12 : hour + 12;
        strHour = String.format("%02d", hour);
        return strHour + strRest;
    }

    public static void main(String[] args) {
        System.out.println(timeConversion("07:05:45AM")); // 07:05:45
        System.out.println(timeConversion("12:00:00AM")); // 00:00:00
        System.out.println(timeConversion("12:00:01AM")); // 00:00:01

        System.out.println(timeConversion("12:00:00PM")); // 12:00:00
        System.out.println(timeConversion("12:45:54PM")); // 12:45:54
        System.out.println(timeConversion("07:05:45PM")); // 19:05:45
    }
}
